src/OpenLoco/Paint/PaintVehicle.cpp

```diff
@@ -0,0 +1,198 @@
+#include "PaintVehicle.h"
+#include "../CompanyManager.h"
+#include "../Config.h"
+#include "../Graphics/Colour.h"
+#include "../Objects/ObjectManager.h"
+#include "../Objects/VehicleObject.h"
+#include "../Things/Vehicle.h"
+#include "Paint.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::Things::Vehicle;
+
+namespace OpenLoco::Paint
+{
+    // 0x00500160
+    const uint8_t _reversePitch[16]{
+        0, 5, 6, 7, 8, 1, 2, 3, 4, 10, 9, 12, 11, 0, 0, 0
+    };
+
+    // 0x004FFAE8
+    const uint32_t _ghostInvisImageFlags[2]{
+        Gfx::recolour(0, PaletteIndex::index_2C),
+        Gfx::recolourTranslucent(0, PaletteIndex::index_31),
+    };
+
+    // 0x004B0CFC
+    static void paintBogie(PaintSession& session, vehicle_bogie* bogie)
+    {
+        auto* vehObject = ObjectManager::get<vehicle_object>(bogie->object_id);
+        if (bogie->object_sprite_type == 0xFF)
+        {
+            return;
+        }
+
+        auto& sprite = vehObject->bogie_sprites[bogie->object_sprite_type];
+        uint8_t yaw = (bogie->sprite_yaw + (session.getRotation() << 4)) & 0x3F;
+        uint8_t pitch = bogie->sprite_pitch;
+
+        if (bogie->getFlags38() & Flags38::isReversed)
+        {
+            yaw ^= (1 << 5);
+            pitch = _reversePitch[bogie->sprite_pitch];
+        }
+        auto yawIndex = (yaw >> 1) & 0x1F;
+
+        switch (pitch)
+        {
+            case 0:
+            case 2:
+            case 9:
+            case 4:
+            case 11:
```

Comment:
Any idea what these represent yet? Should we add constants for them?

Reply:
Yes it's the different states of the bank. I could name them something like 15deg, 30deg,...

Reply:
Sounds good. We use something similar in OpenRCT2, right?

Reply:
```
enum
{
    TRACK_SLOPE_NONE = 0,
    TRACK_SLOPE_UP_25 = 2,
    TRACK_SLOPE_UP_60 = 4,
    TRACK_SLOPE_DOWN_25 = 6,
    TRACK_SLOPE_DOWN_60 = 8,
    TRACK_SLOPE_UP_90 = 10,
    TRACK_SLOPE_DOWN_90 = 18,

    TRACK_VANGLE_TOWER = 10,
    TRACK_VANGLE_REVERSE_FREEFALL = 10
};```
These are the OpenRCT2 constants

Reply:
Think I'll go for:
```
enum class Pitch
{
flat = 0,
12degUp = 1,
25degUp = 2,
45degUp = 3,
60degUp = 4,
12degDown = 5,
25degDown = 6,
45degDown = 7,
60degDown = 8,
}
```

Reply:
Oh actually those are far too steep for this game I'll come up with better names for the angles.